test/serializer-test.ts

```diff
@@ -50,6 +50,92 @@ describe("serializer", function(): void {
     assert.deepEqual(serialized, expectedJSON);
   });
 
+  it("serializes a payment to an X-address with a tag in XRP", function(): void {
+    // GIVEN a transaction which represents a payment to a destination and tag, denominated in XRP.
+    const value = "1000";
+    const destination = "XVfC9CTCJh6GN2x8bnrw3LtdbqiVCUvtU3HnooQDgBnUpQT";
+    const fee = "10";
+    const sequence = 1;
+    const account = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ";
+    const publicKey = "testPublicKey";
+
+    const paymentAmount = new XRPAmount();
+    paymentAmount.setDrops(value);
+
+    const payment = new Payment();
+    payment.setDestination(destination);
+    payment.setXrpAmount(paymentAmount);
+
+    const transactionFee = new XRPAmount();
+    transactionFee.setDrops(fee);
+
+    const transaction = new Transaction();
+    transaction.setAccount(account);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKeyHex(publicKey);
+
+    // WHEN the transaction is serialized to JSON.
+    const serialized = Serializer.transactionToJSON(transaction);
+
+    // THEN the result is as expected.
+    const expectedJSON = {
+      Account: account,
+      Amount: value + "",
+      Destination: "rU6K7V3Po4snVhBBaU29sesqs2qTQJWDw1",
+      DestinationTag: 12345,
+      Fee: fee,
+      Sequence: sequence,
+      TransactionType: "Payment",
+      SigningPubKey: publicKey
+    };
+    assert.deepEqual(serialized, expectedJSON);
+  });
+
+  it("serializes a payment to an X-address without a tag in XRP", function(): void {
+    // GIVEN a transaction which represents a payment to a destination without a tag, denominated in XRP.
+    const value = "1000";
+    const destination = "XVfC9CTCJh6GN2x8bnrw3LtdbqiVCUFyQVMzRrMGUZpokKH";
+    const fee = "10";
+    const sequence = 1;
+    const account = "r9LqNeG6qHxjeUocjvVki2XR35weJ9mZgQ";
+    const publicKey = "testPublicKey";
+
+    const paymentAmount = new XRPAmount();
+    paymentAmount.setDrops(value);
+
+    const payment = new Payment();
+    payment.setDestination(destination);
+    payment.setXrpAmount(paymentAmount);
+
+    const transactionFee = new XRPAmount();
+    transactionFee.setDrops(fee);
+
+    const transaction = new Transaction();
+    transaction.setAccount(account);
+    transaction.setFee(transactionFee);
+    transaction.setSequence(sequence);
+    transaction.setPayment(payment);
+    transaction.setSigningPublicKeyHex(publicKey);
+
+    // WHEN the transaction is serialized to JSON.
+    const serialized = Serializer.transactionToJSON(transaction);
+
+    // THEN the result is as expected.
+    const expectedJSON = {
+      Account: account,
+      Amount: value + "",
+      Destination: "rU6K7V3Po4snVhBBaU29sesqs2qTQJWDw1",
+      DestinationTag: undefined,
```

Comment:
NIT: Not sure if you need to mention the destination tag for the object here.

Reply:
The tests do pass. I don't know how ripple-lib will ultimately handle this.  Perhaps @intelliot  knows? 

Aside, this is one of the reasons I want to move to an mono repo. I can't test this until I upgrade Appian and Xpring-JS to use this code, without setting up `npm link`.

Reply:
I recommend using `npm link` (or `yarn link`) -- it should work fairly well (feel free to let me know if you have issues).

Also, if you'd like to switch to a monorepo, I'm in favor. Go for it.

How is this JSON used? Technically, `undefined` is not a valid JSON value (but it is valid in JavaScript). If the value should be undefined (when parsed), then it should be omitted from the JSON.

Where/how does this get passed to ripple-lib?

Reply:
I apologize, I didn't actually submit my response to your query. 

I use this JSON because it's the format that other ripple libraries seem to want. For instance, to generate a signature, I need to use `ripple-binary-codec`: https://github.com/xpring-eng/xpring-common-js/blob/master/src/signer.ts#L33

When this gets submitted back to Rippled, I also encode back to JSON, encode the signature, and then use the `submit` method: 
https://github.com/xpring-eng/AppianJS/blob/master/src/index.js#L101

If ripplelib and ripplebinary codec will take x-addresses in the destination field, then I can actually drop this PR.